invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/CustomFieldSerializer.js

```diff
@@ -16,28 +16,39 @@ export class CustomFieldSerializer {
     serializedDefault = null,
     allowEmpty = false,
     vocabularyFields = [],
+    genericVocabularies = [],
   }) {
     this.fieldpath = fieldpath;
     this.deserializedDefault = deserializedDefault;
     this.serializedDefault = serializedDefault;
     this.allowEmpty = allowEmpty;
     this.vocabularyFields = vocabularyFields;
+    this.genericVocabularies = genericVocabularies;
   }
 
   #mapCustomFields(record, customFields, mapValue) {
     if (customFields !== null) {
       for (const [key, value] of Object.entries(customFields)) {
         const isVocabularyField = this.vocabularyFields.includes(key);
+        const isGenericVocabulary = this.genericVocabularies.includes(key);
         const _value = _isArray(value)
-          ? value.map((v, i) => mapValue(v, i, isVocabularyField))
-          : mapValue(value, null, isVocabularyField);
+          ? value.map((v, i) => mapValue(v, i, isVocabularyField, isGenericVocabulary))
```

Comment:
I think now the `isVocabularyField` is quite confusing as name because `isGenericVocabulary` should also be a vocabulary field. Do you intend to split between generic and custom?

Reply:
Yes because the custom field we added is a non generic vocabulary, hence why this check was needed.